### PR TITLE
Make Optional things Optional

### PR DESCRIPTION
**Pull request by ihasdapie**
I combed through all 154 matches of = None in the codebase and made Optional things, well, `Optional`

Signed-off-by: Brian Chen <brian.chen@openrobotics.org>

### CODE REVIEW

**Comment by fujitatomoya** [comment]
+1 great effort 👍

File: rclpy/rclpy/__init__.py [hunk 1]
```diff
@@ -181,7 +184,11 @@ def create_node(
         ))
 
 
-def spin_once(node: 'Node', *, executor: 'Executor' = None, timeout_sec: float = None) -> None:
+def spin_once(
+        node: 'Node',
+        *,
+        executor: Optional[Executor] = None,
```
Comment (clalancette) [inline rclpy/rclpy/__init__.py:190]: I'm pretty sure this change (and the related ones below) are what are causing CI to fail.  Maybe:

```suggestion
        executor: Optional['Executor'] = None,
```

will work?

**Review by clalancette — CHANGES_REQUESTED** [review]
This looks generally good to me, with the exception of the failing tests.  Once that is fixed, and CI is green, I'm happy to approve.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18775/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13299/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19516/)

**Review by sloretz — APPROVED** [review]

**Pull request merged**